Assets/Scripts/Audio/FmodEventHandler.cs

```diff
@@ -0,0 +1,335 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+/// <summary>
+/// Class that holds the information for playing and updating parameters on a specific fmod event, for both tonal and atonal events.
+/// The actual playing of the fmod event will occur through the fmod event pool based on the information stored in this class.
+/// Any time we need to play an fmod event, it should be accessed through a gameobject with an instance of this class.
+/// </summary>
+public class FmodEventHandler : MonoBehaviour
+{
+    public string sfxEventName = "";
+    public float sfxEventVolume = 1.0f;
+    public string musicalSfxParamName = "";
+
+    //Used in the GlissUp and GlissDown behaviors to determine spacing between notes
+    public float glissTimeBetweenNotes = 0.1f;
+
+    //Determines what this object does when it is told to play an fmod event
+    public enum TonalSfxMode
+    {
+        //No Tonal Sfx behavior
+        None,
+
+        //Single tonal note behaviors
+        Root,
+        Upwards,
+        Downwards,
+        UpThenDown,
+        DownThenUp,
+        Random,
+
+        //Multiple tonal notes behaviors
+        Chord,
+        GlissUp,
+        GlissDown,
+    };
+
+    public TonalSfxMode sfxMode;
+
+    //Determines what this object's index does upon a chord change
+    public enum IndexBehaviorOnNewChord
+    {
+        DoNothing,
+        ResetNoteToInitIndex,
+        ResetIndexToLastIndexPlayed,
+    };
+
+    public IndexBehaviorOnNewChord indexBehaviorOnNewChord;
+
+    public Rigidbody rb;
+    public bool debugPrintVelocity = false;
+
+    private int noteIndex = 0;
+    private int lastPlayedNoteIndex = 0;
+    private int noteDirection;
```

Comment:
This int is a boolean in disguise. It's only ever -1 or 1.

Reply:
While you're correct, I would prefer to leave this as an integer since simplifies the math of progressing through my chord.